src/common/classes/fb_string.h

```diff
@@ -430,108 +314,217 @@ namespace Firebird
 			return toSize;
 		}
 
-		static unsigned int hash(const_pointer string, const size_type tableSize);
+		virtual void assign(const void* s, size_type n)
+		{
+			baseAssign(s, n);
+		}
+
+	protected:
+		// Following methods are protected to allow to override them in derived classes
+		// without being virtual
 
-		unsigned int hash(size_type tableSize) const
+		void assign(const AbstractString& v)
+		{
+			if (this != &v)
+			{
+				baseAssign(v.c_str(), v.length());
+			}
+		}
+		void assign(const_pointer s)
+		{
+			assign(s, static_cast<size_type>(strlen(s)));
+		}
+		void assign(const size_type n, const char_type c)
 		{
-			return hash(c_str(), tableSize);
+			checkLength(n);
+			memset(getBuffer(n, false), c, n);
 		}
+		// Pick up substring from other string
+		void assign(const AbstractString& v, size_type pos, size_type n = npos);
 
-		bool equalsNoCase(const_pointer string) const;
+		// Move buffer from source if possible
+		void move(AbstractString& v)
+		{
+			if (this != &v)
+			{
+				if (&getPool() == &v.getPool() && v.stringBuffer != v.inlineBuffer) // We can and have what to steal
+				{
+					if (stringBuffer != inlineBuffer) // we also have allocated buffer, exchange
+					{
+						std::swap(stringBuffer, v.stringBuffer);
+						std::swap(stringLength, v.stringLength);
+					}
+					else // Just take it
+					{
+						stringBuffer = v.stringBuffer;
+						stringLength = v.stringLength;
+						v.stringBuffer = v.inlineBuffer;
+						v.stringLength = 0;
+					}
+				}
+				else // fallback to copy
+				{
+					assign(v);
+				}
+			}
+		}
 
-		AbstractString& append(const AbstractString& str)
+		void append(const AbstractString& str)
```

Comment:
These routines are private. They cannot be called from outside code and have nothing to do with public interface of std::basic_string except name.